### PR TITLE
Update readme to include section on missing analysis

### LINKED ISSUE #598: Enable configuration to fail CI if new errors are introduced
We are currently trying to migrate from LGTM to `codeql-action` due to [limitations in LGTM's infrastructure](https://github.community/t/lgtm-extraction-failure-for-c-asp-net-5-0-application/185834). 

With LGTM, we could set it return a failing check any time new analysis alerts were introduced in an PR. This would allow us to prevent merging a PR if it introduced new alerts.

We set up the default GitHub Actions for `codeql-action` and it ran and found 6 alerts in our test suite, but the build did not put up a failing check that would have prevented merging.

This is our first PR that adds `codeql-action`, so perhaps this needs to be merged first so that future PRs have something to be compared against? We are hesitant to merge without knowing there is a way to configure the action to fail when new alerts are introduced.

- PR: https://github.com/sillsdev/TheCombine/pull/1198
- Scanning results: https://github.com/sillsdev/TheCombine/pull/1198/checks?check_run_id=2914899948
- Alerts for PR: https://github.com/sillsdev/TheCombine/security/code-scanning?query=ref%3Arefs%2Fpull%2F1198%2Fmerge+tool%3ACodeQL

### PR DESCRIPTION
**Pull request by aeisenberg**
Fixes #598

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [n/a] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

**Comment by aeisenberg** [comment]
The new text in the readme is mostly copied from the original docs page, which describes the issue so nicely I didn't want to make any changes.

File: README.md [hunk 1]
```diff
@@ -128,3 +128,11 @@ By default, this will override any queries specified in a config file. If you wi
 ## Troubleshooting
 
 Read about [troubleshooting code scanning](https://help.github.com/en/github/finding-security-vulnerabilities-and-errors-in-your-code/troubleshooting-code-scanning).
+
+### Note on "missing analysis" message
+
+The very first time code scanning is run and if it is on a pull tequest, you will probably get a message mentioning a "missing analysis". This is expected.
```
Comment (adityasharad) [inline README.md:134]:
```suggestion
The very first time code scanning is run and if it is on a pull request, you will probably get a message mentioning a "missing analysis". This is expected.
```

File: README.md [hunk 2]
```diff
@@ -128,3 +128,11 @@ By default, this will override any queries specified in a config file. If you wi
...
+
+After code scanning has analyzed the code in a pull request, it needs to compare the analysis of the topic branch (the branch you used to create the pull request) with the analysis of the base branch (the branch into which you want to merge the pull request). This allows code scanning to compute which alerts are newly introduced by the pull request, which alerts were already present in the base branch, and whether any existing alerts are fixed by the changes in the pull request. Initially, if you use a pull request to add code scanning to a repository, the base branch has not yet been analyzed, so it's not possible to compute these details. In this case, when you click through from the results check on the pull request you will see the "Missing analysis for base commit SHA-HASH" message.
```
Comment (adityasharad) [inline README.md:136]: The PR merge commit is analysed rather than the PR branch. Worth mentioning that?

**Review by adityasharad — APPROVED** [review]
Thanks! Minor suggestions.

**Pull request merged**